### ISSUE TITLE
Inject additional properties to the `EditorConfig` interface

### ISSUE THREAD

**Issue by pomek** [ckeditor/ckeditor5#13074]
## 📝 Provide a description of the improvement

TypeScript understands module augmentation that allows injecting new properties into the `EditorConfig` interface.

https://github.com/ckeditor/ckeditor5/blob/e35b6ce10e8bb3d13dfa5640d8287bcb0f13c4ba/packages/ckeditor5-typing/src/typingconfig.ts#L38-L46

Unfortunately, `typedoc` does not process the `declare module { ... }`. We need to find a way to "tell typedoc" that it should inject an interface as a member of `EditorConfig`.

**Comment by pomek** [comment]
Together with @psmyrek, we created a POC. See: https://github.com/ckeditor/ckeditor5-dev/compare/ck/13074?expand=1.

It works with the following source code:

```ts
/**
 * The granularity of undo/redo for typing and deleting. The value `20` means (more or less) that a new undo step
 * is created every 20 characters are inserted or deleted.
 *
 * @memberOf module:core/editor/editorconfig~EditorConfig typing
 */
export interface TypingConfig {
	undoStep?: number;
}
```

**Comment by pomek** [comment]
On a call with @niegowski and @arkflpc we agreed to use the following syntax: `@member module:core/editor/editorconfig~EditorConfig#typing`

**Comment by psmyrek** [comment]
Recently we discussed how to extend interfaces from the outside (from other modules) with `declare module ...` declaration. We wanted to introduce a new `@member` tag, as noted above.

It turns out that it will not be needed. Typedoc knows how to process the module augmentation, but in its own way: it tries to extend the interface where the re-export exists (in `index.ts`), not the one from the source. We managed to fix it on our side in a custom Typedoc plugin.
